packages/registry/src/registry/registry.ts

```diff
@@ -3,14 +3,16 @@ import {ComponentType} from 'react';
 import {ComponentMetadata} from './component-metadata';
 import Metadata from './metadata';
 import {getCompName} from '../utils/get-comp-name';
+import {getCompId} from '../utils/get-comp-id';
```

Comment:
Let's just have `.getId()` on the component metadata.

Reply:
I want to add a `getId()` on the `ComponentMetadata`, but we need a `getCompId` outside of it, so we can check for duplicate ids BEFORE we even create a ComponentMetadata.

Reply:
We should check for duplicate IDs not __before__, but __when__ someone is registering an ID for the component. Since ID is the component's name, and we already have validation for that, you don't need to do anything.

You can simply add `.getId()` on the component metadata and that's it, it's going to work as expected and throw an error if you have components with duplicate names.